apps/dash-uber-rides-demo/app.py

```diff
@@ -0,0 +1,500 @@
+    xVal = []
+    yVal = []
+    xSelected = []
+    colorVal = [
+        "#F4EC15",
+        "#DAF017",
+        "#BBEC19",
+        "#9DE81B",
+        "#80E41D",
+        "#66E01F",
+        "#4CDC20",
+        "#34D822",
+        "#24D249",
+        "#25D042",
+        "#26CC58",
+        "#28C86D",
+        "#29C481",
+        "#2AC093",
+        "#2BBCA4",
+        "#2BB5B8",
+        "#2C99B4",
+        "#2D7EB0",
+        "#2D65AC",
+        "#2E4EA4",
+        "#2E38A4",
+        "#3B2FA0",
+        "#4E2F9C",
+        "#603099",
+    ]
+
+    # Put selected times into a list of numbers xSelected
+    xSelected.extend([int(x) for x in selection])
+
+    for i in range(24):
+        # If bar is selected then color it white
+        if i in xSelected and len(xSelected) < 24:
+            colorVal[i] = "#FFFFFF"
+        xVal.append(i)
+        # Get the number of rides at a particular time
+        yVal.append(len(totalList[month][day][totalList[month][day].index.hour == i]))
+    return [np.array(xVal), np.array(yVal), np.array(colorVal)]
+
+
+# Selected Data in the Histogram updates the Values in the DatePicker
+@app.callback(
+    Output("bar-selector", "value"),
+    [Input("histogram", "selectedData"), Input("histogram", "clickData")],
+)
+def update_bar_selector(value, clickData):
+    holder = []
+    if clickData:
+        holder.append(str(int(clickData["points"][0]["x"])))
+    if value:
+        for x in value["points"]:
+            holder.append(str(int(x["x"])))
+    return list(set(holder))
+
+
+# Clear Selected Data if Click Data is used
+@app.callback(Output("histogram", "selectedData"), [Input("histogram", "clickData")])
+def update_selected_data(clickData):
+    if clickData:
+        return {"points": []}
+
+
+# Update the total number of rides Tag
+@app.callback(Output("total-rides", "children"), [Input("date-picker", "date")])
+def update_total_rides(datePicked):
+    date_picked = dt.strptime(datePicked, "%Y-%m-%d")
+    return "Total Number of rides: {:,d}".format(
+        len(totalList[date_picked.month - 4][date_picked.day - 1])
+    )
+
+
+# Update the total number of rides in selected times
+@app.callback(
+    Output("total-rides-selection", "children"),
+    [Input("date-picker", "date"), Input("bar-selector", "value")],
+)
+def update_total_rides_selection(datePicked, selection):
+    date_picked = dt.strptime(datePicked, "%Y-%m-%d")
+    if selection is None or len(selection) is 0:
+        return ""
+    totalInSelection = 0
+    for x in selection:
+        totalInSelection += len(
+            totalList[date_picked.month - 4][date_picked.day - 1][
+                totalList[date_picked.month - 4][date_picked.day - 1].index.hour
+                == int(x)
+            ]
+        )
+    return "Total rides in selection: {:,d}".format(totalInSelection)
+
+
+# Update Range of List
+@app.callback(
+    Output("date-value", "children"),
+    [Input("date-picker", "date"), Input("bar-selector", "value")],
```

Comment:
You have the same inputs as line 230. Is it possible to use multiple output callbacks?